numba/cuda/cudaimpl.py

```diff
@@ -493,6 +493,134 @@ def ptx_fp16_binary(context, builder, sig, args):
 lower_fp16_binary(stubs.fp16.hmul, 'mul')
 
 
+def inline_asm_helper(builder, func_ret_ty, func_arg_types,
+                      asm_str, constraints_str, asm_args):
+    fnty = ir.FunctionType(func_ret_ty, func_arg_types)
+    asm = ir.InlineAsm(fnty, asm_str, constraints_str)
+    return builder.call(asm, asm_args)
+
+
+@lower(stubs.fp16.hdiv, types.float16, types.float16)
+def lower_fp16_divide(context, builder, sig, args):
+    x, y = args
+
+    # Floating point 16 division algorithm implemented
+    # using Newton Raphson algorithm.
+
+    # x = arg1, y = arg2
+    # Implement x/y
+
+    # Convert fp16 args to fp32 so we can implement
+    # division by reciprocal multiplication on fp32
+    # values
+
+    asm = "cvt.f32.f16 $0, $1;"
+    constr = "=f,h"
+    x_fp32 = inline_asm_helper(builder, ir.FloatType(), [ir.IntType(16)],
+                               asm, constr, [x])
+    y_fp32 = inline_asm_helper(builder, ir.FloatType(), [ir.IntType(16)],
+                               asm, constr, [y])
+
+    # Obtain a gross estimate of the reciprocal of argument 2 (y)
+    # which will be refined. This is the first step in the Newton
+    # Raphson method.
+
+    # y0 = 1/y * (1-err), here y0 is not a precise 1/y
+    # it is an approximation and the term 'err' stands for the
+    # relative error of the approximation.
+
+    yn = inline_asm_helper(builder, ir.FloatType(), [ir.FloatType()],
+                           "rcp.approx.ftz.f32 $0, $1;", "=f,f", [y_fp32])
+
+    # Implement fp32 division via multiplication by
+    # reciprocal of arg2
+
+    # Our objective is to get a quotient x/y, correctly rounded to nearest
+    # even in fp16 format, so we can do q0 = x * yn.
```

Comment:
I've had some difficulty understanding how to interpret this comment: is it "correctly rounded to nearest, even in fp16 format", or is it "correctly rounded nearest even (digit), in fp16 foramt"?